wgengine/netstack/netstack.go

```diff
@@ -285,7 +306,12 @@ func (ns *Impl) Start(lb *ipnlocal.LocalBackend) error {
 	ns.lb = lb
 	// size = 0 means use default buffer size
 	const tcpReceiveBufferSize = 0
-	const maxInFlightConnectionAttempts = 1024
+
+	maxInFlightConnectionAttempts, ok := envknob.LookupInt("TS_DEBUG_NETSTACK_TCP_FWD_MAX_CONNECTIONS")
+	if !ok || maxInFlightConnectionAttempts < 16 || maxInFlightConnectionAttempts > 65536 {
+		maxInFlightConnectionAttempts = 1024
+	}
```

Comment:
Can we pull this out into a separate PR? I want to look at the TCP forwarder code in gVisor a bit more before we give users the ability to configure this... and in general we try to avoid having a "make Tailscale work" envknob if we can instead make it work automatically 😃

Reply:
Yep, agreed, and I do want to eventually support something like this... but once we merge the envknob it's much harder to change, even if it's not an officially-supported thing. Very plausible that this is the right approach, I just didn't want to hold up the metrics changes on that 😅